sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/samples/CreateNewSampleSteps.java

```diff
@@ -0,0 +1,151 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.sormas.e2etests.steps.web.application.samples;
+
+import static org.sormas.e2etests.pages.application.samples.CreateNewSamplePage.*;
+import static org.sormas.e2etests.pages.application.samples.SampleManagementPage.*;
+
+import com.google.common.truth.Truth;
+import cucumber.api.java8.En;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import javax.inject.Inject;
+import org.sormas.e2etests.helpers.WebDriverHelpers;
+import org.sormas.e2etests.pojo.Sample;
+import org.sormas.e2etests.services.SampleService;
+
+public class CreateNewSampleSteps implements En {
+  public static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("M/dd/yyyy");
+  public static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ofPattern("HH:mm");
+  public static Sample sample;
+  public static String sampleId;
+  private final WebDriverHelpers webDriverHelpers;
+
+  @Inject
+  public CreateNewSampleSteps(WebDriverHelpers webDriverHelpers, SampleService sampleService) {
+    this.webDriverHelpers = webDriverHelpers;
+
+    When(
+        "^I create a new Sample with specific data$",
+        () -> {
+          sample = sampleService.buildGeneratedSample();
+          webDriverHelpers.clickOnWebElementBySelector(INTERNAL_IN_HOUSE_TESTING);
+          fillDateOfCollection(sample.getDateOfCollection());
+          fillTimeOfCollection(sample.getTimeOfCollection());
+          selectSampleType(sample.getSampleType());
+          selectReasonForSample(sample.getReasonForSample());
+          fillSampleID(sample.getSampleID());
+          fillCommentsOnSample(sample.getCommentsOnSample());
+          webDriverHelpers.clickOnWebElementBySelector(SAVE_SAMPLE_BUTTON);
+        });
+
+    When(
+        "^I check the created Sample is correctly displayed on Edit Sample page",
+        () -> {
+          final Sample actualSample = collectSampleData();
+          Truth.assertThat(sample).isEqualTo(actualSample);
+        });
+
+    When(
+        "^I change all Sample fields and save$",
+        () -> {
+          sample = sampleService.buildEditSample();
+          webDriverHelpers.clickOnWebElementBySelector(INTERNAL_IN_HOUSE_TESTING);
+          fillDateOfCollection(sample.getDateOfCollection());
+          fillTimeOfCollection(sample.getTimeOfCollection());
+          selectSampleType(sample.getSampleType());
+          selectReasonForSample(sample.getReasonForSample());
+          fillSampleID(sample.getSampleID());
+          fillCommentsOnSample(sample.getCommentsOnSample());
+          webDriverHelpers.clickOnWebElementBySelector(SAVE_EDIT_SAMPLE_BUTTON);
+        });
+
+    When(
+        "I collect the sample UUID displayed on create new sample page",
+        () -> sampleId = collectSampleUuid());
```

Comment:
@BojteTamas  I'm fine with this approach, but this hasn't been used in the other tests. We collected it again in a new POJO. What do you think?

Reply:
We will improve this after I will merge the API branch

Reply:
Agree.